services/thermal_mgr/thermal_mgr.c

```diff
@@ -43,18 +45,47 @@ void initThermalSystemManager(lm75bd_config_t *config) {
 
 error_code_t thermalMgrSendEvent(thermal_mgr_event_t *event) {
   /* Send an event to the thermal manager queue */
-
+  error_code_t errCode;
+  if(event==NULL) return ERR_CODE_INVALID_ARG;
+  if (xQueueSend(thermalMgrQueueHandle,event,portMAX_DELAY)==errQUEUE_FULL) return ERR_CODE_QUEUE_FULL;//task will remain blocked as long as queue is full
   return ERR_CODE_SUCCESS;
 }
 
 void osHandlerLM75BD(void) {
   /* Implement this function */
+  thermal_mgr_event_t event;
+  event.type=THERMAL_MGR_EVENT_INTERRUPT;
+  thermalMgrSendEvent(&event);
 }
 
 static void thermalMgr(void *pvParameters) {
   /* Implement this task */
+  error_code_t errCode;
+  thermal_mgr_event_t event;
+  float temp;
```

Comment:
in general try to limit the scope of your variables to the smallest scope possible